format/FlightSql.proto

```diff
@@ -1550,6 +1755,48 @@ message DoPutUpdateResult {
   int64 record_count = 1;
 }
 
+/*
+ * Request message for the "CancelQuery" action.
+ *
+ * Explicitly cancel a running query.
+ *
+ * This lets a single client explicitly cancel work, no matter how many clients
+ * are involved/whether the query is distributed or not, given server support.
+ * The transaction/statement is not rolled back; it is the application's job to
+ * commit or rollback as appropriate. This only indicates the client no longer
+ * wishes to read the remainder of the query results or continue submitting
+ * data.
+ *
+ * This command is idempotent.
+ */
+message ActionCancelQueryRequest {
+  option (experimental) = true;
+
+  // The result of the GetFlightInfo RPC that initated the query.
+  // XXX(ARROW-16902): this must be a serialized FlightInfo, but is
+  // rendered as bytes because Protobuf does not really support one
+  // DLL using Protobuf definitions from another DLL.
+  bytes info = 1;
```

Comment:
Should this be a serialized flightinfo? or should it actually be a Ticket?

Reply:
It should be a FlightInfo because I assume the server needs the information of _all_ endpoints in order to fully cancel a query. Also in the event that we do update FlightInfo with an application metadata field, it would automatically get passed back to the server

Reply:
Gotcha, that makes sense then.